react/src/StartupService.jsx

```diff
@@ -41,6 +41,46 @@ export default class StartupService {
 
   static mapImslpLink(imslpLink) { // eslint-disable-line
     // TODO Call the API to get the real information.
+    let url = window.location.href;
+    let indexToCutOut = url.indexOf("wiki") + 5;
+    //If Composer
+    if (url.indexOf("Category") >= 0) {
+      indexToCutOut = indexToCutOut + 9;
+      url = url.slice(indexToCutOut);
+      let lastname = url.slice(0,url.indexOf(","));
+      let firstname = url.slice(url.indexOf(",")+2);
+      let title = firstname + '_' +lastname;
+      title = title.replace(new RegExp('_', 'g'), ' ');
+
+      const url = `http://mcmapi.azurewebsites.net/search/entities?q=${title}&limit=1`;
+      return fetch(url, {
+        headers: {
+          'API-KEY': 123,
+        },
+      })
+        .then(res => res.json())
+        .then(res => {
+          return res[0].artists[0];
+        });
+    }
+    //If Composition
+    else {
+      url = url.slice(indexToCutOut);
+      let composition = url.slice(0,url.indexOf('(')-1);
+      composition = composition.replace(new RegExp('_', 'g'), ' ');
+      const title = composition;
+
+      const url = `http://mcmapi.azurewebsites.net/search/entities?q=${title}&limit=1`;
+      return fetch(url, {
+        headers: {
+          'API-KEY': 123,
```

Comment:
You could use the [RemoteService](https://github.com/MusicConnectionMachine/VisualizationG5/blob/develop/react/src/RemoteService.js) instead of fetch.
```
RemoteService.get('/search/entities', new Map([['q', title], ['limit', 1]]))
  .then(JSON.parse)
  .then(...)
```